### PR TITLE
CSM (Vault) integration

### PR DESCRIPTION
**Pull request by dincho**
https://www.pivotaltracker.com/story/show/161711220

### CODE REVIEW

File: packer/scripts/dump-gcp-credentials.sh [hunk 1]
```diff
@@ -1,8 +0,0 @@
-#!/bin/bash
```
Comment (lucafavatella) [inline packer/scripts/dump-gcp-credentials.sh:1]: How can [Makefile](https://github.com/aeternity/epoch-machine-image/blob/5d80d9cd8bb7fc5d37920956560e71721e4f0fac/Makefile#L8) work after deletion of this?

File: .circleci/config.yml [hunk 2]
```diff
@@ -25,11 +25,17 @@ references:
         curl -X POST -H 'Content-Type: application/json' ${ROCKET_HOOK_URL:?} --data "${HOOK_DATA:?}"
       when: on_fail
 
+  setup_ci_environment: &setup_ci_environment
+    run:
+      name: Setup Environment
+      command: echo "source /infrastructure/import-secrets.sh" >> $BASH_ENV
```
Comment (lucafavatella) [inline .circleci/config.yml:31]: I see this uses aeternity/infrastructure image (`infrastructure_config`). Ok.

File: packer/scripts/dump-gcp-credentials.sh (hunk 1 above)
Comment (lucafavatella) [inline packer/scripts/dump-gcp-credentials.sh:1]: Shall README be updated?

File: .circleci/config.yml (hunk 2 above)
Comment (lucafavatella) [inline .circleci/config.yml:31]: Should this not have been run by the Docker entrypoint, hence be available in following packer step, hence making this added step unnecessary? I see nothing before this step that may have changed outcome of script so this seems to me to only have the effect of writing BASH_ENV file as opposed to rely on already-set environment to be accessible to step. Could you please clarify?

File: packer/ansible/image-build.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ---
 - import_playbook: /infrastructure/ansible/setup.yml
 - import_playbook: /infrastructure/ansible/monitoring.yml
+  vars:
+    datadog_api_key: DATADOG_API_KEY
```
Comment (lucafavatella) [inline packer/ansible/image-build.yml]: Do you mean the literal `DATADOG_API_KEY` or the value that the var `DATADOG_API_KEY` has in the environment as set by import-secrets.sh?
I am not sure what is the intended value, and whether this is a specific Ansible syntax.

File: packer/scripts/dump-gcp-credentials.sh (hunk 1 above)
Comment (dincho) [reply packer/scripts/dump-gcp-credentials.sh:1]: Oops, fixed in e912a91

Comment (dincho) [reply packer/scripts/dump-gcp-credentials.sh:1]: Not sure what to update in there?
The image is still build and published to AWS and GCP

File: packer/ansible/image-build.yml (hunk 3 above)
Comment (dincho) [reply packer/ansible/image-build.yml]: It's literal. The idea is that this key is set to some "random" string while building the image. We don't want secrets build-in into the machine images.

Then during the boostrap it's set to a working secret api key. it cannot be set to falsy value unfortunately to be more clear. I changed it a bit d76efa2, I hope it's better now.

File: .circleci/config.yml (hunk 2 above)
Comment (dincho) [reply .circleci/config.yml:31]: CircleCI does not run the entrypoints by default. There are options, but I'll try to improve that in future, now I strive to make it work and stabilize the configuration after the Vault integration damage :)

File: packer/scripts/dump-gcp-credentials.sh (hunk 1 above)
Comment (lucafavatella) [reply packer/scripts/dump-gcp-credentials.sh:1]: Ok thanks.

**Review by lucafavatella — APPROVED** [review]

**Review by valkov-ivaylo — APPROVED** [review]
I cannot spot any other issues than the onces @lucafavatella  mentioned and were addressed.

**Review by acidsennui — APPROVED** [review]

**Pull request merged**